### PR TITLE
[OptionsResolver] Micro optimizations and simplifications

### PR DESCRIPTION
**Pull request by yceruto**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

As we know, this component has a big impact on the workflow of the `Form` component. I'm a newcomer in Blackfire, so there could be other optimizations that I can't able to see, but here we go. 

For now, we've less code to maintain commit-changes and a micro-optimizacion of performance. It'd be great if someone could try these changes in a real project to see their real value.

orbf
metrics

https://blackfire.io/profiles/compare/a04a13d3-7f60-4434-a2b8-0762efb8fbd6/graph
https://github.com/yceruto/orbf The sample takes into account only the core extensions.

Cheers!

### CODE REVIEW

**Comment by yceruto** [comment]
Maybe improving this https://github.com/symfony/symfony/issues/29055

File: src/Symfony/Component/OptionsResolver/OptionsResolver.php [hunk 1]
```diff
@@ -987,58 +987,43 @@ public function offsetGet($option/*, bool $triggerDeprecation = true*/)
...
-                $invalidTypes[$this->formatTypeOf($item, $type).$suffix] = $value;
+        try {
+            if ('null' === $type) {
+                if (null === $value) {
+                    return true;
+                }
+            } elseif (('is_'.$type)($value)) {
+                return true;
+            }
+        } catch (\Error $e) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/OptionsResolver/OptionsResolver.php]: Swallowing any `Error` is very suspicious to me. can we be more specific here? Shouldn't we rethrow? The previous had not such try/catch.

Comment (yceruto) [reply src/Symfony/Component/OptionsResolver/OptionsResolver.php]: I thought about removing all `\function_exists()` calls, but you're right, we could be hiding some real errors from custom `is_*` functions. 

Rolling back to the `\function_exists()` version.

**Review by nicolas-grekas — APPROVED** [review]

**Comment by yceruto** [comment]
Last graph after https://github.com/symfony/symfony/pull/29065#discussion_r231172140:
https://blackfire.io/profiles/compare/96202784-0af8-416f-916d-81544624a068/graph

**Comment by nicolas-grekas** [comment]
Thank you @yceruto.

**Pull request merged**